core/src/raw/http_util/header.rs

```diff
@@ -189,6 +190,45 @@ pub fn parse_into_metadata(path: &str, headers: &HeaderMap) -> Result<Metadata>
     Ok(m)
 }
 
+/// parse_metadata will parse http headers(including standards http headers
+/// and user defined metadata header) into Metadata.
+///
+/// # Arguments
+///
+/// * `user_metadata_prefix` is the prefix of user defined metadata key
+///
+/// # Notes
+///
+/// before return the user defined metadata, we'll strip the user_metadata_prefix from the key
+pub fn parse_metadata(
```

Comment:
Hi, I'd prefer to keep that logic in the service for now. We can extract it later if we find it's common across different services.

Reply:
It may seem like premature optimization, but the user meta prefix is used in other services, such as S3 and GCS. but it's ok to do it later